group1/communication/node/main.js

```diff
@@ -27,30 +28,48 @@ var homedir = (process.platform === 'win32') ? process.env.HOMEPATH : process.en
 // just for file upload
 var done = false;
 app.use(multer({
-        'dest': '../../comm/static/uploads',
+        'dest': '../../comm/static/uploads/',
         onFileUploadStart: function(file) {
+            if (testNameValidation(file.originalname)) {
                 console.log(file.originalname + ' is starting...');
+            } else {
+                console.log('File name is invalid');
+                return false;
+            }
         },
         onFileUploadComplete: function(file) {
                 console.log(file.fieldname + ' uploaded to ' + file.path);
                 done = true;
         }
 }));
 
+function testNameValidation(text) {
+    var format = /[ !@#$%^&*()+\-=\[\]{};':"\\|,<>\/?]/;
+    if (text.trim() == null || text.trim() == "" || format.test(text) == true)  {
+        return false;
+    } else {
+        return true;
+    }
+}
+
 app.post('/upload', function(req,res){
-        if (done==true) {
-                var hash = crypto.randomBytes(20).toString('hex');
-				var directory =  path.join(homedir + '/sites/www.3blueprints.com/static/uploads/');
-                fs.mkdirs(path.join(directory,hash), function(err) {
-                        if(err){
-                            console.log(err);
-                        } else {
-                        var final_path = path.join(hash, req.files.fileUpload.originalname);
-                        fs.rename(req.files.fileUpload.path, directory + final_path);
-                        res.send('static/uploads/' + final_path);
-                        }
-                });
-        }
+    var pathname = url.parse(req.headers.referer).pathname;
+    if (done==true) {
+        var hash = crypto.randomBytes(20).toString('hex');
+		var directory =  path.resolve(process.cwd() + '/../../comm/static/uploads/');
```

Comment:
The spacing still looks off to me on this line.  Line 59 should be at the same indentation level as line 58.